src/utils/fs.rs

```diff
@@ -108,72 +106,41 @@ pub fn copy_files_except_ext(
     }
 
     for entry in fs::read_dir(from)? {
-        let entry = entry?;
+        let entry = entry?.path();
         let metadata = entry
-            .path()
             .metadata()
-            .with_context(|| format!("Failed to read {:?}", entry.path()))?;
+            .with_context(|| format!("Failed to read {entry:?}"))?;
+
+        let entry_file_name = entry.file_name().unwrap();
+        let target_file_path = to.join(entry_file_name);
 
         // If the entry is a dir and the recursive option is enabled, call itself
         if metadata.is_dir() && recursive {
-            if entry.path() == to.to_path_buf() {
+            if entry == to.as_os_str() {
                 continue;
             }
 
             if let Some(avoid) = avoid_dir {
-                if entry.path() == *avoid {
+                if entry == *avoid {
                     continue;
                 }
             }
 
             // check if output dir already exists
-            if !to.join(entry.file_name()).exists() {
-                fs::create_dir(&to.join(entry.file_name()))?;
+            if !target_file_path.exists() {
+                fs::create_dir(&target_file_path)?;
             }
 
-            copy_files_except_ext(
-                &from.join(entry.file_name()),
-                &to.join(entry.file_name()),
-                true,
-                avoid_dir,
-                ext_blacklist,
-            )?;
+            copy_files_except_ext(&entry, &target_file_path, true, avoid_dir, ext_blacklist)?;
         } else if metadata.is_file() {
             // Check if it is in the blacklist
-            if let Some(ext) = entry.path().extension() {
+            if let Some(ext) = entry.extension() {
                 if ext_blacklist.contains(&ext.to_str().unwrap()) {
                     continue;
                 }
             }
-            debug!(
```

Comment:
I removed this trace as it did not really add value, in my opinion. The target file path is traced also below.